api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+        that is invalid with respect to the schema's type.
+
+        * Referring any of Validation, Enrichment or Partitioning strategies that does not exist or whose
+        parameterization is deemed invalid.
+
+        Nakadi MIGHT impose necessary schema, validation and enrichment minimal configurations that MUST be
+        followed by all EventTypes (examples include: validation rules to match the schema; enriching every
+        Event with the reception date-type; adhering to a set of schema fields that are mandatory for all
+        EventTypes). **The mechanism to set and inspect such rules is not defined at this time and might not
+        be exposed in the API.**
+
+      parameters:
+        - name: event-type
+          in: body
+          description: EventType to be created
+          schema:
+            type: object
+            $ref: '#/definitions/EventType'
+          required: true
+      responses:
+        '200':
+          description: Ok
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '422':
+          description: Unprocessable Entity
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+
+
+  /event-types/{name}:
+    get:
+      tags:
+        - schema-registry-api
+      description: |
+        Returns the `EventType` identified by its name.
+      parameters:
+        - name: name
+          in: path
+          description: Name of the EventType to load.
+          type: string
+          required: true
+        - name: X-Flow-Id
+          in: header
+          description: |
+            The flow id of the request, which is written into the logs and passed to called services. Helpful
+            for operational troubleshooting and log analysis.
+          type: string
+          format: flow-id
+      responses:
+        '200':
+          description: Ok
+          schema:
+            $ref: '#/definitions/EventType'
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: EventType not found
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+
+    put:
+      tags:
+        - schema-registry-api
+      description: |
+        Updates the `EventType` identified by its name. Behaviour is the same as creation of `EventType` (See
+        POST /event-type) except where noted below.
+
+        The name and the topic fields cannot be changed. Attempting to do so will result in a 422 failure.
+
+        At this moment changes in the schema are not supported and will produce a 422 failure. (TODO: define
+        conditions for backwards compatible extensions in the schema)
```

Comment:
To be clear: schema evolution IS needed and WILL be supported. Although backwards compatible changes are definitely recommended, they are not always possible and this is clear. The mechanism with which this will be supported in Nakadi (basically we're talking here about versioning of EventTypes/schemas) is what was not defined until now. 

I'll adjust the wording in the API to make this situation more clear.